driver-cql-shaded/src/main/java/io/nosqlbench/activitytype/cql/statements/core/CQLSessionCache.java

```diff
@@ -320,7 +320,7 @@ public void afterChannelInitialized(SocketChannel channel) throws Exception
         Cluster cl = builder.build();
 
         // Apply default idempotence, if set
-        activityDef.getParams().getOptionalBoolean("defaultidempotence").map(
+        activityDef.getParams().getOptionalBoolean("defaultidempotence").ifPresent(
```

Comment:
This one could also impact workloads, but should also be made correct. I'll take notes on these through this review to add to the release notes.

Reply:
these changes were just cosmetic in that we use the right method (that doesnt return a value we wont use)... previous code still had the same sideeffect i think.

Reply:
I believe that the Java streaming semantics mean that the map doesn't cause the stream to be evaluated. I tested with this:
```Java
System.out.println("numbers:");
Stream.of(1,2,3).map(v->{System.out.println("v:"+v); return v;});
System.out.println("letters:");
Stream.of("a","b","c").map(v->{System.out.println("v:"+v); return v;}).collect(Collectors.toList());

```
and got this output:
```
numbers:
letters:
v:a
v:b
v:c
```

Reply:
the type here is `Optional` not `Stream` so:
```
    private static Object printStuff(String s)
    {
        System.out.println(s);
        return null;
    }

    public static void main(String[] args)
    {
        Optional.of("foo").map(s -> printStuff(s));
        Optional.<String>empty().map(s -> printStuff(s));
        Optional.of("bar").ifPresent(s -> printStuff(s));
        Optional.<String>empty().ifPresent(s -> printStuff(s));
    }
```
prints:
```
foo
bar
```

Reply:
Oh, right!